### ISSUE TITLE
[EfficientDet] Train a custom dataset with pretrained weights

### ISSUE THREAD

**Issue by SadiaAfrinPurba** [Tessellate-Imaging/Monk_Object_Detection#38]
Hello,

I have trained a custom dataset from scratch using EfficientDet-D0. The performance was not enough satisfactory. 
Is there any way I can train the custom dataset with pretrained weights (COCO weights) and fine-tune the last layer?

**Comment by abhi-kumar** [comment]
In the code here, 
https://github.com/Tessellate-Imaging/Monk_Object_Detection/blob/master/4_efficientdet/lib/train.py

modify it to load your own weights, freeze base layers, and use it as transfer learning. 

Else you may check up https://github.com/Tessellate-Imaging/Monk_Object_Detection/tree/master/example_notebooks/1_gluoncv_finetune pipeline, it's pure transfer learning.

**Comment by abhi-kumar** [comment]
Feature has been added.

**Comment by abhi-kumar** [comment]
The following features have been added.

a) Support for 8 new models
  - model_name="efficientnet-b0"
  - model_name="efficientnet-b1"
  - model_name="efficientnet-b2"
  - model_name="efficientnet-b3"
  - model_name="efficientnet-b4"
  - model_name="efficientnet-b5"
  - model_name="efficientnet-b6"
  - model_name="efficientnet-b7"
  - model_name="efficientnet-b8"

#Usage 
`gtff.Model(model_name="efficientnet-b0");`

b) Load a previously trained model and start a new training
gtf.Model(model_name="efficientnet-b0", load_pretrained_model_from="path to model.pth");

**Comment by sadransh** [comment]
@abhi-kumar Thanks for your work. 
What is b8? I have not seen that in the main article. 
Are there any validation set AP results for newly added models ?
